lib/jxl/dec_modular.cc

```diff
@@ -274,44 +274,35 @@ Status ModularFrameDecoder::DecodeGroup(const Rect& rect, BitReader* reader,
     Rect r(rect.x0() >> fc.hshift, rect.y0() >> fc.vshift,
            rect.xsize() >> fc.hshift, rect.ysize() >> fc.vshift, fc.w, fc.h);
     if (r.xsize() == 0 || r.ysize() == 0) continue;
-    Channel gc(r.xsize(), r.ysize());
-    gc.hshift = fc.hshift;
-    gc.vshift = fc.vshift;
-    gi.channel.emplace_back(std::move(gc));
-  }
-  if (zerofill) {
-    int gic = 0;
-    for (c = beginc; c < full_image.channel.size(); c++) {
-      Channel& fc = full_image.channel[c];
-      int shift = std::min(fc.hshift, fc.vshift);
-      if (shift > maxShift) continue;
-      if (shift < minShift) continue;
-      Rect r(rect.x0() >> fc.hshift, rect.y0() >> fc.vshift,
-             rect.xsize() >> fc.hshift, rect.ysize() >> fc.vshift, fc.w, fc.h);
-      if (r.xsize() == 0 || r.ysize() == 0) continue;
+    if (zerofill && use_full_image) {
       for (size_t y = 0; y < r.ysize(); ++y) {
         pixel_type* const JXL_RESTRICT row_out = r.Row(&fc.plane, y);
         memset(row_out, 0, r.xsize() * sizeof(*row_out));
       }
-      gic++;
+    } else {
+      Channel gc(r.xsize(), r.ysize());
+      if (zerofill) ZeroFillImage(&gc.plane);
+      gc.hshift = fc.hshift;
+      gc.vshift = fc.vshift;
+      gi.channel.emplace_back(std::move(gc));
     }
-    return true;
   }
+  if (zerofill && use_full_image) return true;
   ModularOptions options;
-  if (!ModularGenericDecompress(
-          reader, gi, /*header=*/nullptr, stream.ID(frame_dim), &options,
-          /*undo_transforms=*/-1, &tree, &code, &context_map))
+  if (!zerofill && !ModularGenericDecompress(reader, gi, /*header=*/nullptr,
+                                             stream.ID(frame_dim), &options,
+                                             /*undo_transforms=*/-1, &tree,
+                                             &code, &context_map))
```

Comment:
Braces pls

Reply:
Updated. Also changed the `&&` to two ifs, to make it a bit clearer that short-circuit evaluation is important.